### PR TITLE
DHFPROD-5799: added keyboard navigation to overview page, header, footer

### PR DESCRIPTION
**Pull request by jyangML**
### Description
https://project.marklogic.com/jira/browse/DHFPROD-5799
#### Changes added
(See comment below for file-by-file changelog)
* header
  * enabled tabbing and arrow keys in header
  * added focus and focus outlines to logo, title, service details, help, and user elements in header
  * changed link location of logo in header to marklogic.com homepage
  * various minor style changes
* overview
  * enabled tabbing and arrow keys in overview cards
  * added focus and focus outlines to all cards
  * moved render to before toolbar for tab ordering
* footer
  * added focus outline to privacy button
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by jyangML** [comment]
Update: forgot to run lint

**Comment by jyangML** [comment]
Oops, running lint seems to have also changed e2e/cypress/integration/login/authorization.spec.tsx.  There's no functional changes, just all the single quotes into double quotes.

**Review by wooldridge — CHANGES_REQUESTED** [review]
@jyangML, really cool to see all this working!

I'm seeing issues now with the regular mouse-based interaction with the User button in the header. Using the mouse, I can't logout since the menu disappears when I try to move to click it, see:

https://drive.google.com/file/d/1E-0e3GliI-peMD8ZWApkS4dVmsmnkZMe/view

For the large boxes, I can tab and shift-tab through all five boxes but with the keyboard arrows I can't get to Run.

Is UX OK with the focus outlines appearing on regular mouse clicks? For example, when I click Data Hub Central, the focus outline appears and persists. I don't expect that to happen. 

As a comparison in the wild, when I go to http://www.google.com, I can tab to About, Store, and the Google logo, and an outline appears (just like ours!). But if I mouse click any of those elements, I don't get the outline. Is there a way to get that sort of interaction?

**Comment by xnikhil08** [comment]
> Oops, running lint seems to have also changed e2e/cypress/integration/login/authorization.spec.tsx. There's no functional changes, just all the single quotes into double quotes.

@jyangML - I have already fixed the e2e/cypress/integration/login/authorization.spec.tsx related linting errors in my PR https://github.com/marklogic/marklogic-data-hub/pull/4979.

**Comment by jyangML** [comment]
Latest changes:
* removed opening and closing user logout menu triggered by mouse hover
* removed focus outline in header elements when clicked

File: marklogic-data-hub-central/ui/src/components/header/header.tsx [hunk 1]
```diff
@@ -141,25 +215,54 @@ const Header:React.FC<Props> = (props) => {
...
+    if (event.key === "ArrowLeft") { logoRef.current!.focus(); }
+  };
+
   return (
     <>
       <Layout.Header className={styles.container}>
-        <div className={styles.logoContainer} aria-label="header-logo" onClick={handleHomeClick}>
-          <Avatar className={styles.logo} src={logo} />
+        <div className={styles.logoContainer} aria-label="header-logo">
+          <a href="https://www.marklogic.com/" aria-label="logo-link" className={styles.logo} tabIndex={1} ref={logoRef}
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui/src/components/header/header.tsx:255]: We've updated the header logo to link to marklogic.com. There's a test in App.test.tsx (line 46) that checks that this goes to the Overview page when clicked. I think this needs to be updated to test the new link, it's failing in my env.

**Review by wooldridge — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/pages/Overview.test.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from "react";
 import {Router} from "react-router";
 import {createMemoryHistory} from "history";
+import userEvent from "@testing-library/user-event";
 import {render, fireEvent} from "@testing-library/react";
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui/src/pages/Overview.test.tsx:5]: I'm also seeing this import for fireEvent repeated on line 10 and my testing is complaining, remove the one on line 10?

**Review by wooldridge — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/pages/Overview.test.tsx (hunk 2 above)
Comment (wooldridge) [reply marklogic-data-hub-central/ui/src/pages/Overview.test.tsx:5]: I see the test file getting fireEvent from two different places, /react and /dom.

Comment (jyangML) [reply marklogic-data-hub-central/ui/src/pages/Overview.test.tsx:5]: fixed

File: marklogic-data-hub-central/ui/src/components/header/header.tsx (hunk 1 above)
Comment (jyangML) [reply marklogic-data-hub-central/ui/src/components/header/header.tsx:255]: fixed

**Comment by jyangML** [comment]
* added some comments for changing the logout dropdown in the future
* fixed unit tests that were passing

**Comment by jyangML** [comment]
There are cypress test failures.  I'll be looking into them.

**Comment by briantang** [comment]
@jyangML You may want to convert the PR to draft state if there is more work to be done.

**Comment by jyangML** [comment]
Fixed some issues with cypress tests failing

**Comment by jyangML** [comment]
Ran ESLint on all the files in this commit

**Review by briantang — APPROVED** [review]
Keyboard navigation forward, backwards, and enter works great. Seeing all tests pass as well.

**Review by briantang — CHANGES_REQUESTED** [review]
@jyangML Noticed that the header tooltips have become unaligned with this PR. Is it straightforward to keep them consistent like before, or was this necessary due to keyboard outline/focus? If it is the latter, we may need to reconsider how to position the tooltips.

Screen Shot 2020-12-16 at 12 50 35 PM
Screen Shot 2020-12-16 at 12 50 19 PM
Screen Shot 2020-12-16 at 12 50 15 PM

**Comment by jyangML** [comment]
Fixed tooltip alignment and spacing

**Review by wooldridge — APPROVED** [review]
This is really great, I can now keyboard nav all the global elements and the homepage. Unit tests pass and look good.

**Comment by wooldridge** [comment]
I'm seeing the tooltips OK as well.

**Review by xnikhil08 — APPROVED** [review]

**Review by briantang — APPROVED** [review]
Looks good now.

**Pull request merged**